src/reactions/render.php

```diff
@@ -38,140 +54,179 @@
 // Generate a unique ID for the block.
 $block_id = 'activitypub-reactions-block-' . wp_unique_id();
 
-$reactions = array();
+// Determine display style.
+$is_summary = 'summary' === $attributes['displayStyle'];
 
-foreach ( Comment::get_comment_types() as $_type => $type_object ) {
-	$_comments = get_comments(
-		array(
-			'post_id' => $_post_id,
-			'type'    => $_type,
-			'status'  => 'approve',
-			'parent'  => 0,
-		)
-	);
+// Build inner content based on display style.
+ob_start();
 
-	if ( empty( $_comments ) ) {
-		continue;
+if ( $is_summary ) {
+	// Collect counts for summary view.
+	$counts = array();
+
+	// Add comment count if enabled.
+	if ( $attributes['showComments'] ) {
+		$comment_count = get_comments_number( $_post_id );
+		if ( $attributes['showEmpty'] || $comment_count > 0 ) {
+			$counts['comments'] = array(
+				'count' => $comment_count,
+				'label' => _n( 'Comment', 'Comments', $comment_count, 'activitypub' ),
+			);
+		}
 	}
 
-	$count = count( $_comments );
-	// phpcs:disable WordPress.WP.I18n
-	$label = sprintf(
-		_n(
-			$type_object['count_single'],
-			$type_object['count_plural'],
-			$count,
-			'activitypub'
-		),
-		number_format_i18n( $count )
-	);
-	// phpcs:enable WordPress.WP.I18n
-
-	$reactions[ $_type ] = array(
-		'label' => $label,
-		'count' => $count,
-		'items' => array_map(
-			function ( $comment ) {
-				return array(
-					'name'   => html_entity_decode( $comment->comment_author ),
-					'url'    => $comment->comment_author_url,
-					'avatar' => get_avatar_url( $comment ),
-				);
-			},
-			$_comments
-		),
-	);
-}
+	// Add reaction counts.
+	foreach ( Comment::get_comment_types() as $_type => $type_object ) {
+		$count = Interactions::count_by_type( $_post_id, $_type );
 
-if ( empty( $reactions ) ) {
-	echo '<!-- Reactions block: No reactions found. -->';
-	return;
-}
+		if ( ! $attributes['showEmpty'] && 0 === $count ) {
+			continue;
+		}
 
-// Set up the Interactivity API config.
-wp_interactivity_config(
-	'activitypub/reactions',
-	array(
-		'defaultAvatarUrl' => ACTIVITYPUB_PLUGIN_URL . 'assets/img/mp.jpg',
-		'namespace'        => ACTIVITYPUB_REST_NAMESPACE,
-	)
-);
+		// phpcs:disable WordPress.WP.I18n
+		$label = sprintf(
+			_n(
+				$type_object['count_single'],
+				$type_object['count_plural'],
```

Comment:
Hm, that's unfortunate. Looks like the translation strings only accept numeric values, so we can't pass a string here. As a result, the number will not be bold here, as it is in the editor. I also just noticed there's a difference in capitalization.

<img width="301" height="104" alt="Image" src="https://github.com/user-attachments/assets/cf9e5b6c-4a1c-42e5-acab-e7d1fe57f81e" />

<img width="348" height="131" alt="Image" src="https://github.com/user-attachments/assets/382d4a93-2a65-4924-a729-a8e404bb5d08" />